docs/RFCS/20200401_alter_column_type.md

```diff
@@ -0,0 +1,189 @@
+This results in unexpected behaviour for code that currently depend on ColumnIDs for ordering. 
+For example in information_schema.columns, ordinal_position is ColumnID.
+This results in SHOW COLUMNS being out of order as it currently depends on ordinal_position which is currently ColumnID.
+
+One solution is adding a "ordinal position" slice field to the TableDescriptor to handle the "logical id" of a column for virtual tables. As it currently stands, this ordinal position value can correspond to the Column ID, it would only change when doing this column swap, so when a swap is performed, the ordinal positions of the columns change. We can store ordinal position in the ColumnDescriptor or TableDescriptor. If we use a slice for OrdinalPositions in table descriptor, each index can correspond to the Columns field in TableDescriptor, thus when doing a swap, between two columns, we wouldn't have to change the OrdinalPositions.
+
+[This page](https://wiki.postgresql.org/wiki/Alter_column_position) mentions how ALTER COLUMN POSITION could be implemented in Postgres. THe popular method suggests using a separate logical and physical identifier for the columns. The ordinal number of the columns would correspond to the logical identifier.
+
+## State when the columns have been swapped but the old column hasn't been dropped.
+During this state, both the pre-swap and post-swap versions of the TableDescriptor are valid, meaning writes to the new column have to be reflected in the old column.
+Until all nodes have the version of the TableDescriptor where the swap has happened, we have to do inserts on both the old and new columns. 
+To solve this problem, we make the old column c a computed column of the new column c'.
+
+This brings up another issue where c is a computed column of c', but a value inserted into c' may not necessarily be able to be casted back to the data type of c.
+
+Example: When a column is converted from int -> string, but the value inserted cannot be converted back to int.
+- For example, if we insert 'hello' into the converted column from int -> string while the old int column has not been dropped yet, it will give a parse error since it will try to parse 'hello' as an int to support insert into the old column.
+
+### USING EXPRESSION inverse problem
+This problem is even trickier the user does the ALTER COLUMN ... TYPE USING EXPRESSION.
+
+If the user used an expression to alter the type, it may be impossible to convert back to the old type (need an inverse function.)
+If one node performs a read from the table with the old TableDescriptor after one node does an insert into the table with a new version of the TableDescriptor, what value can we give for the read?
+
+For example, if we're converting a column from int to bool and using the expression f(x) = true if x > 0 and false otherwise, how do we get an inverse for this function?
+
+### Possible Solutions
```

Comment:
[nit]: it would probably be clearer to just group the possible solutions / pros cons together into lists rather than markdown headers